streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -217,7 +217,7 @@ private String ensureOneNodeIsReady(final List<Node> nodes) {
                 }
             }
             try {
-                kafkaClient.poll(500, Time.SYSTEM.milliseconds());
+                kafkaClient.poll(0, Time.SYSTEM.milliseconds());
```

Comment:
A spinning loop while the connection is being established doesn't seem right to me. If you're using TLS, for example, it will have to do the handshake before the connection is ready.